### ISSUE TITLE
Proposal for `waitForTransaction` improvement

### ISSUE THREAD

**Issue by Cussone** [starknet-io/starknet.js#1237]
Now that block times are much faster it would make sense to lower the `retryInterval` param from `5000ms` (waited on twice, so it's `10s`) to something more realistic.

Also, I find it weird that `retryInterval` is an optional param. In my opinion, it would make more sense to require it explicitly, with a suggested value in the docs. I see it as a better DX.

This would be a breaking change. So, it is something to consider for the next major release.

**Comment by Cussone** [comment]
> hello @Cussone can i hop on this issue?

Not sure what you are referring to, but feel free to write anything you think about this.

**Comment by PhilippeR26** [comment]
Has been discussed in Slack: will not be modified, as user can customize the value.

**Comment by Cussone** [comment]
> Has been discussed in Slack: will not be modified, as user can customize the value.

Yup, that was the conclusion for the current version, but I opened this proposal for the next version, as we also discussed on Slack.

**Comment by CanutoIX** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by Sekhmet** [comment]
Another improvements would be:
- do not call `sleep` after receipt is successfully resolved [here](https://github.com/starknet-io/starknet.js/blob/3bdee6797a72053a699e363388a8d24275e1f389/src/channel/rpc_0_7.ts#L360) - right now even if transaction is ready we are still waiting for 5s.
- do not call `sleep` on entry [here](https://github.com/starknet-io/starknet.js/blob/3bdee6797a72053a699e363388a8d24275e1f389/src/channel/rpc_0_7.ts#L304)

I've had issue with my tests where everything was super slow (even when using starknet-devnet) and actually the issue is that each call to `waitForTransaction` will take at least 10s unless you modify `transactionRetryIntervalFallback`.

**Comment by PhilippeR26** [comment]
Closed, as:
- If you want to go fast (with a single and fast access node, no events usage), you  can use fastExecute() function.
- For other cases (slow nodes, parallelized nodes, slow infra), the 5 s delay is necessary, especially is you declare/deploy/use events. If you are confident that your specific case/infra can go faster, you can customize the retry interval.
